### PR TITLE
feat(patch-request-content-type): add new spectral-style rule

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit introduces the new spectral-style
'patch-request-content-type' rule.
This rule will check to make sure that PATCH operations
use the correct mime-types within their request bodies.

Signed-off-by: Phil Adams <phil_adams@us.ibm.com>

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Dependencies have been updated as needed
- [x] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [x] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [x] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [ ] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [ ] Removed or adjusted testcases (packages/validator/test)
- [ ] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [ ] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/ruleset/test/property-attributes.test.js [hunk 1]
```diff
@@ -145,11 +145,13 @@ describe('Spectral rule: property-attributes', () => {
         };
 
         const results = await testRule(ruleId, rule, testDocument);
-        expect(results).toHaveLength(3);
```
Comment (padamstx) [inline packages/ruleset/test/property-attributes.test.js:148]: this testcase needed to be updated because of changes I made to the rootDocument testcase

**Review by hudlow — APPROVED** [review]

**Review by dpopp07 — APPROVED** [review]
Looks good! 👍

**Pull request merged**